test/src/test/java/org/corfudb/integration/LogReplicationDynamicStreamIT.java

```diff
@@ -0,0 +1,386 @@
+        );
+        writeToMap(mapBStandby, false, 0, numWrites);
+
+        // Start log replication
+        startLogReplicatorServers();
+
+        // Verify mapA is successfully replicated
+        verifyDataOnStandby(mapAStandby, numWrites);
+        // Verify old entries of mapA are cleared on Sink
+        try (TxnContext txn = corfuStoreStandby.txn(CORFU_SYSTEM_NAMESPACE)) {
+            for (int i = 2 * numWrites; i < 2 * numWrites + numWrites / 2; i++) {
+                StringKey key = StringKey.newBuilder().setKey(String.valueOf(i)).build();
+                CorfuStoreEntry<StringKey, IntValue, Metadata> entry = txn.getRecord(mapAStandby, key);
+                assertThat(entry.getPayload()).isNull();
+            }
+            txn.commit();
+        }
+        // Verify mapB on Sink is cleared
+        assertThat(mapBStandby.count()).isEqualTo(0);
+    }
+
+    /**
+     * This test will verify the streams to stream tags map is correctly rebuilt during
+     * snapshot sync.
+     *
+     * (1) Open table with TAG_ONE on Source and Sink
+     * (2) Initiate a testing stream listener and subscribe to TAG_ONE on Sink
+     * (3) Write some entries to the table on Source
+     * (4) Start log replication snapshot sync and verify the test listener received all the updates
+     */
+    @Test
+    public void testStandbyStreamingSnapshotSync() throws Exception {
+        // Open testing map on Source and Sink
+        String streamName = "TableStreaming001";
+
+        Table<StringKey, IntValue, Metadata> mapTagOne = corfuStoreActive.openTable(
+                NAMESPACE,
+                streamName,
+                StringKey.class,
+                IntValue.class,
+                Metadata.class,
+                TableOptions.fromProtoSchema(ValueFieldTagOne.class)
+        );
+
+        Table<StringKey, IntValue, Metadata> mapTagOneStandby = corfuStoreStandby.openTable(
+                NAMESPACE,
+                streamName,
+                StringKey.class,
+                IntValue.class,
+                Metadata.class,
+                TableOptions.fromProtoSchema(ValueFieldTagOne.class)
+        );
+
+        // Subscribe the testing stream listener
+        UUID streamId = CorfuRuntime.getStreamID(mapTagOne.getFullyQualifiedTableName());
+        // The plus one is for the clear entry before applying the stream updates
+        int expectedMessageSize = numWrites + 1;
+        CountDownLatch streamingStandbySnapshotCompletion = new CountDownLatch(expectedMessageSize);
+        StreamingStandbyListener listener = new StreamingStandbyListener(streamingStandbySnapshotCompletion,
+                Collections.singleton(streamId));
+        corfuStoreStandby.subscribeListener(listener, NAMESPACE, DefaultLogReplicationConfigAdapter.TAG_ONE);
+
+        writeToMap(mapTagOne, true, 0, numWrites);
+
+        // The stream to tags map is rebuilt upon receiving the replicated data
```

Comment:
another good test would be to open this map on the Source only.  That will test that the table is replicated on the Sink alongwith the right tags, even though this information was not available on the Sink

Reply:
Discussed offline - stream listener will determine the tables to listen at the time streaming task is created, so for now this case couldn't happen / be tested.

Reply:
I thought a bit more on this.. we can still test it by just verifying that the tags were updated.  No need to attach a listener but just verify the tags after replication

Reply:
Adding another approach for testing what Pankti has suggested: What about creating a stream listener after the map has been replicated to the sink?
You'll need to capture the tail before at the sink before replication and then after the map has been replicated start listening from the captured ts. 
Would this be possible?